src/Main.java

```diff
@@ -0,0 +1,66 @@
+import java.util.ArrayList;
+import java.util.List;
+
+abstract class Animal {
+    private String name;
+
+    public Animal(String name) {
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public abstract String makeSound();
+
+    public void displaySound() {
+        System.out.println(getName() + " makes sound: " + makeSound());
+    }
+}
+
+class Dog extends Animal {
+    public Dog() {
+        super("Dog");
```

Comment:
un Dog este intotdeauna un "Dog"; daca maine vin si modific in
public Dog() {
        super("This is not a Dog - my PO forced me to");